src/main/java/net/barrage/school/java/ecatalog/web/ProductController.java

```diff
@@ -36,10 +37,15 @@ public class ProductController {
     private final Counter listProductsCounter = meterRegistry
             .counter("ecatalog.products.listProducts");
 
+    @Getter(value = AccessLevel.PRIVATE, lazy = true)
+    private final Timer listProductsTimer = meterRegistry
+            .timer("ecatalog.products.listProductsTimer");
+
     @GetMapping
-    @PreAuthorize("hasRole('ROLE_SYSTEM_ADMIN')")
     public List<Product> listProducts() {
+        Timer.Sample sample = Timer.start(meterRegistry);
         var products = productService.listProducts();
```

Comment:
What will happen if `productService.listProducts()` throws an exception. Will your timer ever ends? Better to cover such timers with `try-finally` blocks.